internal/cmd/server/start_alarm_notification_server.go

```diff
@@ -0,0 +1,367 @@
+			ctx,
+			"Failed to create metrics wrapper",
+			slog.String("error", err.Error()),
+		)
+		return exit.Error(1)
+	}
+
+	// Create the router:
+	router := mux.NewRouter()
+	router.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		service.SendError(w, http.StatusNotFound, "Not found")
+	})
+	router.MethodNotAllowedHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		service.SendError(w, http.StatusMethodNotAllowed, "Method not allowed")
+	})
+	router.Use(metricsWrapper, authenticationWrapper, authorizationWrapper)
+
+	// create k8s client with kube(from env first)
+	//var config *rest.Config
+	kubeClient, err := k8s.NewClient().SetLogger(logger).SetLoggingWrapper(loggingWrapper).Build()
+
+	if err != nil {
+		logger.ErrorContext(
+			ctx,
+			"Failed to create kubeClient",
+			"error", err,
+		)
+		return exit.Error(1)
+	}
+
+	// Get the resource server details:
+	resourceServerURL, err := flags.GetString(resourceServerURLFlagName)
+	if err != nil {
+		logger.ErrorContext(
+			ctx,
+			"Failed to get resource server URL flag",
+			"flag", resourceServerURLFlagName,
+			"error", err.Error(),
+		)
+		return exit.Error(1)
+	}
+	if resourceServerURL == "" {
+		logger.ErrorContext(
+			ctx,
+			"Resource server URL is empty",
+			"flag", resourceServerURLFlagName,
+		)
+		return exit.Error(1)
+	}
+	resourceServerToken, err := flags.GetString(resourceServerTokenFlagName)
+	if err != nil || resourceServerToken == "" {
+		logger.ErrorContext(
+			ctx,
+			"Resource server URL is empty",
+			"flag", resourceServerTokenFlagName,
+		)
+		return exit.Error(1)
+	}
+
+	// Create the handler:
+	// Get the namespace:
+	o2imsNamespace, err := flags.GetString(namespaceFlagName)
+	if err != nil {
+		logger.DebugContext(
+			ctx,
+			"Failed to get o2ims namespace flag",
+			"flag", o2imsNamespace,
+			"error", err.Error(),
+		)
+	}
+
+	if o2imsNamespace == "" {
+		o2imsNamespace = service.DefaultNamespace
+	}
+	// Get the configmapName:
+	subscriptionsConfigmapName, err := flags.GetString(subscriptionConfigmapNameFlagName)
+	if err != nil {
+		logger.DebugContext(
+			ctx,
+			"Failed to get alarm subscription configmap name flag",
+			"flag", subscriptionsConfigmapName,
+			"error", err.Error(),
+		)
+	}
+
+	if subscriptionsConfigmapName == "" {
+		subscriptionsConfigmapName = service.DefaultAlarmConfigmapName
+	}
+
+	handler, err := service.NewAlarmNotificationHandler().
+		SetLogger(logger).
+		SetLoggingWrapper(loggingWrapper).
+		SetCloudID(cloudID).
+		SetKubeClient(kubeClient).
+		SetNamespace(o2imsNamespace).
+		SetConfigmapName(subscriptionsConfigmapName).
+		SetResourceServerURL(resourceServerURL).
+		SetResourceServerToken(resourceServerToken).
+		Build(ctx)
+
+	if err != nil {
+		logger.Error(
+			"Failed to create handler",
+			"error", err,
+		)
+		return exit.Error(1)
+	}
+
+	router.HandleFunc(
+		"/o2ims-infrastructureMonitoring/{version}/alarmNotifications",
+		handler.ProcessPost,
```

Comment:
I think this handler should implement the `SeveHTTP` method, and then here it should be like this:

```go
router.Handle(
        "...",
        handler,
)
```